### ISSUE TITLE
VirtualMachineImages getImage with empty version generates ClassCastException

### ISSUE THREAD

**Issue by karldmoore** [Azure/azure-libraries-for-java#377]
Whilst the method prevents passing in null values it allows empty String.  If you do this it generates a ClassCastException way down in Jackson deserialization as it doesn't expect an ArrayNode to be provided.  It would make sense to improve the validation of the input parameters.

**Comment by anuchandy** [comment]
Yes, agreed. We will add an extra validation in the library layer.
